src/Components/Components/src/ComponentFactory.cs

```diff
@@ -14,46 +15,76 @@ internal sealed class ComponentFactory
     private const BindingFlags _injectablePropertyBindingFlags
         = BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic;
 
-    private static readonly ConcurrentDictionary<Type, Action<IServiceProvider, IComponent>> _cachedInitializers = new();
+    private static readonly ConcurrentDictionary<Type, (Action<IServiceProvider, IComponent> propertyInitializers, ObjectFactory? componentInitializer)> _cachedInitializers = new();
```

Comment:
While it looks nicer to shove for the DefaultComponentActivator to have it's own cache for component instantiators, it results in a small regression when lots of components are involved. Dictionary lookups are somewhat slow in WebAssembly and co-locating the two avoids this regression at the cost of unseemly code.

Reply:
+1 on avoiding any perf regression!

Maybe the readability would be a small bit improved by defining some `internal readonly struct ComponentInitializer` just as a named equivalent to this lengthy tuple.

However I don't feel strongly because it's very self-contained within `ComponentFactory.cs` which we don't have to touch very often.

Reply:
Strictly speaking, instead of checking on each call whether or not `_componentActivator` is null and then branching, we could cache the relevant action for each type. That might also simplify the dictionary type signature since the `ObjectFactory` would only exist as something captured within the closure.

Again, I don't feel strongly here because you've shown there's no perf problem with this implementation, and it's not like we have to squeeze every drop of clarity out of this class which is self-contained and we almost never touch.

Reply:
I like this idea a lot because it reduces the cache value to a single `Func`. But it is gnarly with the component activator case. Our current implementation allows you to return different kinds of `IComponent` per-invocation since the cache key is based on the returned type. Consequently this is legal and the implementation will correctly initialize properties:

```C#
var component1 = componentActivator.CreateInstance(typeof(Index)); // -> Index component
var component2 = componentActivator.CreateInstance(typeof(Index)); // -> Counter component
```

This is difficult to support if our cache key was instead the requested component type as would be if we're caching a `Func`